### PR TITLE
Added support for running the webpack dev server in ssl mode

### LINKED ISSUE #1576: Errors using Angular-CLI locally with --ssl true
> Please provide us with the following information:
> ---------------------------------------------------------------

1. OS?  Windows 7, 8 or 10. Linux (which distribution). Mac OSX (Yosemite? El Capitan?)

Windows 10 Enterprise

2. Versions.
angular-cli: 1.0.0-beta.10
node: 6.2.2
os: win32 x64

3. Repro steps. Was this an app that wasn't created using the CLI? What change did you
   do on your code? etc.

I created keys using makecert.exe and openssl to create the .key and .crt files.  
I successfully set up a node server.js file to use those keys to serve an https web site.  To set things in up node, I needed to add a 'passphrase' parameter like so:  

var app = express();
var options = {
    key: fs.readFileSync(__dirname + "\\ssl\\server.pem"),
    cert: fs.readFileSync(__dirname + "\\ssl\\server.crt"),
    passphrase: "mypassphrase"
};
https.createServer(options, app).listen(4001, function () {
    console.log('Https server running on 4001...');
});

But when I use the same keys in ng server --ssl true, I get this error: 

Error: error:0907B068:PEM routines:PEM_READ_BIO_PRIVATEKEY:bad password read

I get the error whether I use the .pem file or the .key file, and there's no 'passphrase' parameter.

4. Stack Trace
    at Error (native)
    at Object.createSecureContext (_tls_common.js:88:19)
    at Server (_tls_wrap.js:749:25)
    at new Server (https.js:26:14)
    at Object.exports.createServer (https.js:47:10)
    at Server.configure (D:\georg\Documents\Visual Studio 2015\Projects\RCxAuth\RCX_AUTH\node_modules\angular-cli
\node_modules\tiny-lr\lib\server.js:49:27)
    at new Server (D:\georg\Documents\Visual Studio 2015\Projects\RCxAuth\RCX_AUTH\node_modules\angular-cli\node_
modules\tiny-lr\lib\server.js:36:8)
    at createServer (D:\georg\Documents\Visual Studio 2015\Projects\RCxAuth\RCX_AUTH\node_modules\angular-cli\nod
e_modules\angular-cli\lib\tasks\server\livereload-server.js:19:14)
    at Class.module.exports.Task.extend.liveReloadServer (D:\georg\Documents\Visual Studio 2015\Projects\RCxAuth\
RCX_AUTH\node_modules\angular-cli\node_modules\angular-cli\lib\tasks\server\livereload-server.js:41:30)
    at Class.module.exports.Task.extend.listen (D:\georg\Documents\Visual Studio 2015\Projects\RCxAuth\RCX_AUTH\n
ode_modules\angular-cli\node_modules\angular-cli\lib\tasks\server\livereload-server.js:46:23)
    at Class.module.exports.Task.extend.start (D:\georg\Documents\Visual Studio 2015\Projects\RCxAuth\RCX_AUTH\no
de_modules\angular-cli\node_modules\angular-cli\lib\tasks\server\livereload-server.js:89:17)
    at Class.module.exports.Task.extend.run (D:\georg\Documents\Visual Studio 2015\Projects\RCxAuth\RCX_AUTH\node
_modules\angular-cli\node_modules\angular-cli\lib\tasks\serve.js:56:24)
    at D:\georg\Documents\Visual Studio 2015\Projects\RCxAuth\RCX_AUTH\node_modules\angular-cli\node_modules\angu
lar-cli\lib\commands\serve.js:64:24
    at lib$rsvp$$internal$$tryCatch (D:\georg\Documents\Visual Studio 2015\Projects\RCxAuth\RCX_AUTH\node_modules
\angular-cli\node_modules\rsvp\dist\rsvp.js:1036:16)
    at lib$rsvp$$internal$$invokeCallback (D:\georg\Documents\Visual Studio 2015\Projects\RCxAuth\RCX_AUTH\node_m
odules\angular-cli\node_modules\rsvp\dist\rsvp.js:1048:17)
    at lib$rsvp$$internal$$publish (D:\georg\Documents\Visual Studio 2015\Projects\RCxAuth\RCX_AUTH\node_modules\
angular-cli\node_modules\rsvp\dist\rsvp.js:1019:11)

4. The log given by the failure. Normally this include a stack trace and some
   more information.

5. Mention any other details that might be useful.

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

### PR DESCRIPTION
**Pull request by sbaramov**
Added support for running the webpack dev server in ssl mode through 'ng serve' command. The command options --ssl, --ssl-key, and --ssl-cert are used to configure the server.

Fixes #1576.

### CODE REVIEW

**Comment by sbaramov** [comment]
i signed it

File: packages/angular-cli/custom-typings.d.ts [hunk 1]
```diff
@@ -17,6 +17,9 @@ interface IWebpackDevServerConfigurationOptions {
   headers?: { [key: string]: string };
   stats?: { [key: string]: boolean };
   inline: boolean;
+  https: boolean;
```
Comment (hansl) [inline packages/angular-cli/custom-typings.d.ts]: This field should probably be optional.

**Comment by sbaramov** [comment]
@hansl , I am sorry but I do not see the e2e test for serve-webpack.ts. Can you give me some pointers?

**Comment by filipesilva** [comment]
@sbaramov the closest test I can find is the one for the `proxy` functionality, it also needs a file and does a request. See `./tests/e2e/tests/misc/proxy.ts`

**Comment by zackarychapple** [comment]
This looks like a great feature, I would only request that you add another param for certificate authority. Both GoDaddy and Symantec have a ca that can go with the crt and key.

**Comment by sbaramov** [comment]
@filipesilva Thanks, the test is implemented now.

**Comment by Brocco** [comment]
@sbaramov PR looks fine, just rebased it for merging & cla got out of whack, can you please comment again that you've signed it?

**Comment by sbaramov** [comment]
Yes i have signed the CLA.

**Comment by Brocco** [comment]
@sbaramov apparently losing the CLA signing is not something we can recover from on this PR, can you please submit the same code in a new PR?

Just tag me in it and I'll take a look. Take care to rebase prior to pushing the new one too, please.

**Comment by sbaramov** [comment]
Closing the pull request in favor of #2792 because the CLA did not work here.

**Pull request closed without merging**